Slider/SliderBehaviorDecorator.js

```diff
@@ -40,63 +41,33 @@ const defaultConfig = {
 const SliderBehaviorDecorator = hoc(defaultConfig, (config, Wrapped) => {
 	const {emitSpotlightEvents} = config;
 
-	return class extends Component {
-		static displayName = 'SliderBehaviorDecorator';
-
-		static propTypes = {
-			activateOnSelect: PropTypes.bool,
-			'aria-valuetext': PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
-			max: PropTypes.number,
-			min: PropTypes.number,
-			orientation: PropTypes.string,
-			value: PropTypes.number
-		};
-
-		static defaultProps = {
-			max: 100,
-			min: 0,
-			orientation: 'horizontal'
-		};
-
-		constructor (props) {
-			super(props);
-
-			this.paused = new Pause();
-			this.handleActivate = this.handleActivate.bind(this);
-			this.handleBlur = this.handleBlur.bind(this);
-			this.handleDragEnd = this.handleDragEnd.bind(this);
-			this.handleDragStart = this.handleDragStart.bind(this);
-			this.handleFocus = this.handleFocus.bind(this);
-			this.handleSpotlightEvents = this.handleSpotlightEvents.bind(this);
-			this.bounds = {};
-			this.sliderRef = createRef();
-
-			this.state = {
-				active: false,
-				dragging: false,
-				focused: false,
-				useHintText: true,
-				prevValue: props.value
-			};
-		}
-
-		static getDerivedStateFromProps (props, state) {
-			if (props.value !== state.prevValue) {
-				return {
-					useHintText: false,
-					prevValue: props.value
-				};
+	const SliderBehavior = (props) => {
+		const sliderBehaviorProps = setDefaultProps(props, sliderDefaultProps);
+
+		const [paused] = useState(() => new Pause());
```

Comment:
why did we need useState instead of just use a variable?

Reply:
I used it to not create new instances every rerender, but I replaced it with useMemo.